guide/miscellaneous/making-an-eval-command.md

```diff
@@ -0,0 +1,110 @@
+
+In other words, eval is dangerous and if somehow someone else has access to it, the person can run (malicious) code with full access.
+Furthermore, anyone with the access to it can get the app's secrets and credentials - such as bot token, API keys, so on.
+
+:::warning
+Note: eval() is also slower than the alternatives, since it has to invoke the JS interpreter, while many other constructs are optimized by modern JS engines.
+:::
+
+## Implementation
+
+### Securing eval
+
+Basically, we will be limiting the eval command only to the owner.
+To do that, we will add an condition, which will return if the author's ID is not equals to owner's ID.
+
+<!-- eslint-skip -->
+
+```js
+if (message.author.id !== 'ownerID') return;
+```
+
+Additionally, you can add your ID in the config file:
+
+```json
+{
+    "prefix": "!",
+    "token": "your-token-goes-here",
+    "owner": "Owner-ID-here"
+}
+```
+
+<!-- eslint-skip -->
+
+```js
+const { owner } = require('../config.json');
+if (message.author.id !== owner) return;
+```
+
+### Cleaning eval
+
+We'll be creating a function that prevents the use of actual mentions within the return line by adding a zero-width character between the @ and the first character of the mention - blocking the mention from happening.
+
+<!-- eslint-skip -->
+
+```js
+const clean = content => {
+    if (typeof content === 'string') {
+        return content.replace(/`/g, '`' + String.fromCharCode(8203)).replace(/@/g, '@' + String.fromCharCode(8203));
+    } else {
+        return content;
+    }
+}
+```
+
+### Object to String representation
+
+We can convert object to string representation by using an in-built Node.js module - `util`. Specifically, `util#inspect()`
+So what is `util#inspect`? `util.inspect()` method returns a string representation of object that is intended for debugging.
+Read more on [Node.js API Docs](https://nodejs.org/api/util.html#util_util_inspect_object_options).
+
+First, requiring the `util` module, however, we just need the inspect module, so we will simply destruct it from the module.
+
+```js
+const { inspect } = require('util');
+```
+
+Second, checking if the result is a string or not, if it is, inspect it:
+
+<!-- eslint-skip -->
+
+```js
+let res = eval(args.join(' '));
+if (typeof res !== 'string') res = inspect(res);
+```
+
+## Conclusion
+
+Here's the final code with enhancements.
+
+<!-- eslint-skip -->
+
+```js
+const { inspect } = require('util');
+const { owner } = require('../config.json');
+
+if (message.author.id !== owner) return;
+try {
+    let res = eval(args.join(' '));
+    if (typeof res !== 'string') res = inspect(res);
```

Comment:
Not sure about default depth personally. It defaults to 2, and that is a lot, considering that we can reach and list all guild's information like name and icon from inspecting `message.member`. That drastically decreases chances of getting anything back except API Error about exceeding message length.